src/awkward/_broadcasting.py

```diff
@@ -688,14 +689,14 @@ def continuation():
                 for x in inputs
             ):
                 # Ensure all layouts have same length
-                length = None
+                length = unset
                 for x in inputs:
                     if isinstance(x, Content):
-                        if length is None:
+                        if length is unset:
                             length = x.length
                         elif backend.nplike.known_shape:
                             assert length == x.length
-                assert length is not None
+                assert length is not unset
```

Comment:
I'm uneasy about how we handle unknown lengths here. There's a lot to consider in broadcasting, and I've not spent a lot of time in my mind on it. I'm hopeful that this PR mostly just renames `UnknownLength`, and therefore we can review this logic down the road.